src/watcher.js

```diff
@@ -66,8 +65,7 @@ export function proxyFileChanges(
   {artifactsDir, onChange, filePath, shouldWatchFile}: ProxyFileChangesParams
 ): void {
   if (!shouldWatchFile) {
```

Comment:
I say just remove this and make `shouldWatchFile` mandatory. It's not needed for anything except the tests (which can be fixed up). I tried this change locally so I think it will work:
````diff
diff --git a/src/watcher.js b/src/watcher.js
index 83ff86f..c42bf04 100644
--- a/src/watcher.js
+++ b/src/watcher.js
@@ -19,7 +19,7 @@ export type OnSourceChangeParams = {|
   sourceDir: string,
   artifactsDir: string,
   onChange: OnChangeFn,
-  shouldWatchFile?: ShouldWatchFn,
+  shouldWatchFile: ShouldWatchFn,
 |};
 
 // NOTE: this fix an issue with flow and default exports (which currently
@@ -59,7 +59,7 @@ export type ProxyFileChangesParams = {|
   artifactsDir: string,
   onChange: OnChangeFn,
   filePath: string,
-  shouldWatchFile?: ShouldWatchFn,
+  shouldWatchFile: ShouldWatchFn,
 |};
 
 export function proxyFileChanges(
````